tests/PHPStan/Analyser/nsrt/narrow-bool-cast.php

```diff
@@ -27,3 +27,13 @@ function doFoo(string $x, array $arr): void {
 	}
 	assertType('array{}|array{string}', $matches);
 }
+
+
+/** @param int<-5, 5> $x */
+function castInt($x) {
```

Comment:
I couldn't think of a better test-sample. 

I think testing type-inference will get more meaningful examples, after we understand loose comparisons better